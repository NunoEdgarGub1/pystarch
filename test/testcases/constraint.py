

def f(a, b, c):
    d = a + b
    e = d - 4
    return e


def g(x):
    return -x

def h(x):
    return x + 1
