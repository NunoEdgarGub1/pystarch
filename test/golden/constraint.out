f Function(a: Union(Num,Str,List(Unknown)), b: Union(Num,Str,List(Unknown)), c: Unknown -> Num)
g Function(x: Num -> Num)
h Function(x: Num -> Num)
j Function(x: Union(List(Unknown),Dict(Unknown,Unknown),BaseTuple) -> Unknown)
k Function(a: Str, b: Num -> Num)

testcases/constraint.py:9 type-error "d" (Union(Num,Str,List(Unknown)) vs Num)
